api/controllers/accounts.js

```diff
@@ -43,74 +84,93 @@ function AccountsController(scope) {
  * @param {function} next
  * @todo Add description for the function and the params
  */
-AccountsController.getAccounts = function(context, next) {
+AccountsController.getAccounts = async function(context, next) {
 	const params = context.request.swagger.params;
 
 	let filters = {
-		address: params.address.value,
-		publicKey: params.publicKey.value,
-		secondPublicKey: params.secondPublicKey.value,
-		username: params.username.value,
+		address_eql: params.address.value,
+		publicKey_eql: params.publicKey.value,
+		secondPublicKey_eql: params.secondPublicKey.value,
+		username_like: params.username.value,
+	};
+
+	const options = {
 		limit: params.limit.value,
 		offset: params.offset.value,
 		sort: params.sort.value,
+		extended: true,
 	};
 
 	// Remove filters with null values
 	filters = _.pickBy(filters, v => !(v === undefined || v === null));
 
-	modules.accounts.shared.getAccounts(_.clone(filters), (err, data) => {
-		if (err) {
-			return next(err);
-		}
+	try {
+		let lastBlock = await storage.entities.Block.get(
+			{},
+			{ sort: 'height:desc', limit: 1 }
+		);
+		lastBlock = lastBlock[0];
 
-		data = _.cloneDeep(data);
-
-		data = _.map(data, account => {
-			if (_.isEmpty(account.delegate)) {
-				delete account.delegate;
-			} else {
-				account.delegate.rank = parseInt(account.delegate.rank);
-			}
-			if (_.isNull(account.publicKey)) {
-				account.publicKey = '';
-			}
-			if (_.isNull(account.secondPublicKey)) {
-				account.secondPublicKey = '';
-			}
-			delete account.secondSignature;
-			delete account.unconfirmedSignature;
-			return account;
-		});
+		const data = await storage.entities.Account.get(filters, options).map(
+			accountFormatter.bind(
+				null,
+				lastBlock.height ? blockReward.calcSupply(lastBlock.height) : 0
```

Comment:
how can lastBlock.height be 0?

Reply:
It can't be zero, its just a check if the `lastBlock` exists or not. Copied exactly from the previous code.